Shared.fsx

```diff
@@ -519,25 +519,31 @@ let iNameToEhList =
                        else None)
                 |> List.ofArray
             | None -> []
+        if ownEventHandler.Length > 0 then ownEventHandler else []
 
+    allInterfaces
+    |> Array.map (fun i -> (i.Name, GetEventHandler i))
+    |> Map.ofArray
+
+let iNameToEhParents =
+    let hasHandler (i : Browser.Interface) =
+        iNameToEhList.ContainsKey i.Name && not iNameToEhList.[i.Name].IsEmpty
+
+    // Get all the event handlers from an interface and also from its inherited / implemented interfaces
+    let rec GetEventHandler(i : Browser.Interface) =
         let extendedEventHandler =
             match GetInterfaceByName i.Extends with
-            | Some i -> GetEventHandler i
-            | None -> []
+            | Some i when hasHandler i -> [i]
+            | _ -> []
 
         let implementedEventHandler =
             let implementis = i.Implements |> Array.map GetInterfaceByName
             [ for i' in implementis do
                   yield! match i' with
-                         | Some i -> GetEventHandler i
+                         | Some i ->  if hasHandler i then [i] else []
```

Comment:
Then the name of the variable should be changed to `implementedInterface` and other places, otherwise it can be confusing later